### PR TITLE
ViewViewComponentResult doesn't handle `~/` in view paths correctly

### LINKED ISSUE #2856: ViewViewComponentResult doesn't handle `~/` in view paths correctly
View Components ignore the `~` if you're trying to specify a full path to a view.

Ex:

return View("~/Widgets/MyComponent/MyView.cshtml");

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #2856

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak

File: src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/ViewViewComponentResult.cs [hunk 1]
```diff
@@ -113,5 +116,8 @@ private static IViewEngine ResolveViewEngine(ViewComponentContext context)
         {
             return context.ViewContext.HttpContext.RequestServices.GetRequiredService<ICompositeViewEngine>();
         }
+
+        private static bool IsApplicationRelativePath(string path) =>
+            path[0] == '~' || path[0] == '/';
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/ViewViewComponentResult.cs]: 12 PranavPoints awarded. Seriously this is harder to read than the normal function syntax. Please stop, we're worried about you :crying_cat_face:

Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/ViewViewComponentResult.cs]: Suggest inlining into the condition above. As-is looks like a source of `IndexOutOfRangeException`s.

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/ViewViewComponentResultTest.cs [hunk 2]
```diff
@@ -275,22 +275,54 @@ public void Execute_DoesNotWrapThrownExceptionsInAggregateExceptions()
...
+        {
+            // Arrange
+            var viewEngine = new Mock<ICompositeViewEngine>(MockBehavior.Strict);
+            viewEngine
+                .Setup(v => v.FindPartialView(It.IsAny<ActionContext>(), viewName))
+                .Returns(ViewEngineResult.Found(string.Empty, new Mock<IView>().Object))
+                .Verifiable();
+            var viewData = new ViewDataDictionary(new EmptyModelMetadataProvider());
+            var componentContext = GetViewComponentContext(new Mock<IView>().Object, viewData);
+            var expectedViewName = $"Components/{componentContext.ViewComponentDescriptor.ShortName}/Default";
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/ViewViewComponentResultTest.cs]: ???

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/ViewViewComponentResult.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/ViewViewComponentResult.cs]: :+1:

**Pull request closed without merging**